runners/flink/src/main/java/org/apache/beam/runners/flink/metrics/FlinkMetricContainerBase.java

```diff
@@ -84,14 +88,59 @@ public void updateMetrics(String stepName, List<MetricsApi.MonitoringInfo> monit
    * given step.
    */
   void updateMetrics(String stepName) {
-    MetricResults metricResults = asAttemptedOnlyMetricResults(metricsContainers);
+    List<String> stepNameList =
+        Arrays.asList(stepName, GlobalMetricsUtils.GLOBAL_CONTAINER_STEP_NAME);
+    MetricResults metricResults =
+        asAttemptedOnlyMetricResultsForSteps(metricsContainers, stepNameList);
     MetricQueryResults metricQueryResults =
-        metricResults.queryMetrics(MetricsFilter.builder().addStep(stepName).build());
+        metricResults.queryMetrics(
+            MetricsFilter.builder()
+                .addStep(stepName)
+                .addStep(GlobalMetricsUtils.GLOBAL_CONTAINER_STEP_NAME)
+                .build());
     updateCounters(metricQueryResults.getCounters());
     updateDistributions(metricQueryResults.getDistributions());
     updateGauge(metricQueryResults.getGauges());
   }
 
+  /**
+   * Similar to {@link MetricsContainerStepMap#asAttemptedOnlyMetricResults}, it gets the metrics
+   * results from the MetricsContainerStepMap. Instead of getting from all steps, it gets result
+   * from only interested steps. Thus, it's more efficient.
+   */
+  private static MetricResults asAttemptedOnlyMetricResultsForSteps(
```

Comment:
It may make sense to put this under `MetricsContainerStepMap`. Either place should be fine.